tools/run_tests/filter_pull_request_tests.py

```diff
@@ -104,7 +105,7 @@ def add_trigger(self, trigger):
   'config\.m4$': [_PHP_TEST_SUITE],
   'CONTRIBUTING\.md$': [],
   'Gemfile$': [_RUBY_TEST_SUITE],
-  'grpc.def$': [_WINDOWS_TEST_SUITE],
+  'grpc\.def$': [_WINDOWS_TEST_SUITE],
   'grpc\.gemspec$': [_RUBY_TEST_SUITE],
```

Comment:
One case that these definitions are missing is when someone manually updates files that are supposed to be autogenerated.  (e.g.  files under /vsprojects).  In that case, sanity tests won't run, yet from my experience people often forget that some files are generated and update them manually - and we want to prevent that from happening.

You probably need to add sanity test run for all updates to generated files (which is somewhat tricky, because there is a lot of them and more can be added later - without having a way to enumerate them, we might need to just add _SANITY_TEST_SUITE to all whitelist entries)
E.g.
`'^vsprojects/': [_WINDOWS_TEST_SUITE, _SANITY_TEST_SUITE],`

Actually, I don't understand why sanity takes such long time to run - I think we should be running it as often as possible.